### ISSUE TITLE
Don't test release in CI

### ISSUE THREAD

**Issue by aatifsyed** [ChainSafe/forest#3021]
I recently had a half hour CI job:
https://github.com/ChainSafe/forest/actions/runs/5312623318/jobs/9618981175?pr=3012

Here's the time breakdown
![image](https://github.com/ChainSafe/forest/assets/38045910/a9f86188-87f6-4f09-bd3d-1f6aadb13da5)

~60% of the time is in the release tests.
If tests pass in debug, but fail in release, that's a compiler bug.
We don't need to test the compiler, we can just remove these tests.

(Or at least, make them a parallel job)

**Comment by lemmih** [comment]
Benefits of testing with `--release`: The test environment more closely matches how the final product is being executed.

Drawbacks: Compiling with `--release` is slow.

Given that @aatifsyed has done a great job at removing compilation feature flags, the compilation time might not be significant anymore. The latest run finished in about 8 minutes. @aatifsyed, are you okay with leaving in the `--release` tests if we can keep the total runtime under 10 minutes?

**Comment by ruseinov** [comment]
Perhaps we could also look at ways to speed up builds that require 3rd party tools. Though 8 min is not all that bad I guess - it’s still pretty slow as the project isn’t very big either.

Also, I wonder if building cli and daemon together made a positive impact on the outcome, even though they were being built in parallel before.

**Comment by aatifsyed** [comment]
I'm ok with leaving `--release` in, but I think our CI is a bit of a mess, I'll raise a new Issue

**Comment by LesnyRumcajs** [comment]
@ruseinov No, the impact of this is negative. Even with all the feature flags, the duration was around six minutes. 

https://github.com/ChainSafe/forest/actions/workflows/unit-tests.yml?page=10&query=is%3Asuccess

Then it downgraded to >15 minutes and timing out from time to time.
https://github.com/ChainSafe/forest/actions/workflows/unit-tests.yml?page=2&query=is%3Asuccess

Now it's back to reasonable bounds without the feature flags but I'm pretty sure it could be faster.
